test-app/app/services/storage.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  type ServiceFactory,
+  provide,
+  service,
+  singleton,
+} from 'ember-polaris-service';
+import Storage from 'ember-storage';
+import { enable } from 'ember-storage/extensions';
+import { type StorageWithBatching } from 'ember-storage-batch';
+
+export class AppStorage extends Storage implements StorageWithBatching {
+  #storage = service(this, singleton(localStorage));
+
+  async getItem(key: string): Promise<string | null> {
+    return this.getItemSync(key);
+  }
+
+  async putItem(key: string, value: string): Promise<void> {
+    this.putItemSync(key, value);
+  }
+
+  async deleteItem(key: string): Promise<void> {
+    this.deleteItemSync(key);
+  }
+
+  // StorageWithBatching
+  async batchGet(keys: string[]): Promise<(string | null)[]> {
+    return keys.map((key) => this.getItemSync(key));
+  }
+
+  async batchUpdate(
+    items: [key: string, value: string | null][],
+  ): Promise<void> {
+    for (const [key, value] of items) {
+      if (value === null) {
+        this.deleteItemSync(key);
+      } else {
+        this.putItemSync(key, value);
+      }
+    }
+  }
+
+  // Custom Extensions
+  getItemSync(key: string): string | null {
+    return this.#storage.getItem(this.keyFor(key));
+  }
+
+  hasItemSync(key: string): boolean {
+    return this.getItemSync(key) !== null;
+  }
+
+  putItemSync(key: string, value: string): void {
+    this.#storage.setItem(this.keyFor(key), value);
+  }
+
+  deleteItemSync(key: string): void {
+    this.#storage.removeItem(this.keyFor(key));
+  }
+
+  *entries(): Generator<[string, string], void, void> {
```

Comment:
This is accessed in the JSONStorage service as proof that these custom extensions can be used in the app just fine